server/routes/datasources/datasources_router.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { schema } from '@osd/config-schema';
+import { IRouter } from '../../../../../src/core/server';
+import { DATASOURCES_BASE } from '../../../common/constants/shared';
+
+export function registerDatasourcesRoute(router: IRouter) {
+  router.get(
+    {
+      path: `${DATASOURCES_BASE}/{name}`,
+      validate: {
+        params: schema.object({
+          name: schema.string(),
+        }),
+      },
+    },
```

Comment:
Do we want an OSD APIs for this. I thought this is calling a SQL/PPL endpoint with a query to fetch datasources.

Reply:
Yes - not sure what you are getting at here?